src/pages/FolderPage/FolderPage.tsx

```diff
@@ -0,0 +1,185 @@
+import { ChangeEvent, useState } from "react";
+import { NoResults } from "pages";
+import styles from "./folder.module.css";
+import Header from "./components/Header";
+import CardList from "components/CardList/CardList";
+import SearchBar from "components/Input/SearchBar/SearchBar";
+import { ReactComponent as AddIcon } from "assets/images/ic_add.svg";
+import DeleteModal from "components/Modal/DeleteModal/DeleteModal";
+import SharedModal from "components/Modal/SharedModal/SharedModal";
+import FolderModal from "components/Modal/FolderModal/FolderModal";
+import Category from "./components/Category/Category";
+import ActionButton from "./components/ActionButton/ActionButton";
+import useModal from "utils/hooks/useModal";
+import {
+  ALL,
+  DELETE_FOLDER,
+  ADD_FOLDER,
+  SHARED,
+  EDIT,
+} from "utils/constants/strings";
+import { GetLinkResponse, GetFolderResponse } from "types/apis";
+import { useGetFolders } from "utils/hooks/useGetFolders";
+import { useGetLinks } from "utils/hooks/useGetLinks";
+
+type Nullable<T> = T | null;
+
+export type SelectedCategory = {
+  id: Nullable<number>;
+  name: string;
+};
+
+export type ButtonClick = (
+  categoryId: Nullable<number>,
+  categoryName: string
+) => void;
+
+interface UseFetchResponse<T> {
+  data: Nullable<T>;
+  loading?: boolean;
+  error?: string;
+}
+
+function FolderPage() {
+  let userId = null;
+
+  const value = localStorage.getItem("userId");
+
+  if (typeof value === "string") {
+    userId = JSON.parse(value);
+  }
+
+  const [selectedCategory, setSelectedCategory] = useState<SelectedCategory>({
+    id: null,
+    name: ALL,
+  });
+
+  const { modals, openModal, closeModal } = useModal();
+
+  const { data: folders }: UseFetchResponse<GetFolderResponse[]> =
+    useGetFolders(userId);
+
+  const { data: folderLinks }: UseFetchResponse<GetLinkResponse[]> =
+    useGetLinks(userId, selectedCategory.id);
+
+  const [searchInput, setSearchInput] = useState("");
+
+  const [searchParam] = useState(["url", "title", "description"]);
+
+  const searchItmes = (e: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(e.target.value);
+  };
+
+  const filterSearchText = (items: GetLinkResponse[]) => {
```

Comment:
함수가 하는 일이 검색어로 아이템을 필터링하는 것 같은데요~
지금은 검색어를 필터링하는 것처럼 느껴져서 아래와 같이 표현해주면 어떨까 싶네요!
```suggestion
  const filterItemsWithSearchText = (items: GetLinkResponse[]) => {
```

Reply:
확인해보니까 함수로 만들필요없이 그대로 서버에서 받아온 값을 가공해줘도 괜찮을 것 같아요~
대신에 이렇게 하려면 folderLinks가 배열 타입으로 정의되어 있어야 할 것 같네요!

```ts
  const filteredItems = folderLinks?.filter((item) => {
    return searchParam.some((newItem) => {
      return (
        item[newItem] &&
        item[newItem]
          .toString()
          .toLowerCase()
          .indexOf(searchInput.toLowerCase()) > -1
      );
    });
  });
```